frame/wormhole/backing/s2s/src/lib.rs

```diff
@@ -267,6 +283,43 @@ pub mod pallet {
 			T::PalletId::get().into_account()
 		}
 	}
+
+	impl<T: Config> MessageConfirmer for Pallet<T> {
+		fn on_messages_confirmed(message_id: BridgeMessageId, result: bool) -> Weight {
+			let (user, token) = <LockedQueue<T>>::take(message_id);
+			if !result {
+				let token_info = match &token {
+					Token::Native(info) => {
+						log::debug!("cross receive native token {:?}", info);
+						info
+					}
+					Token::Erc20(info) => {
+						log::debug!("cross receive erc20 token {:?}", info);
+						return 1;
+					}
+					_ => {
+						log::debug!("unrecognized token type");
+						return 1;
+					}
+				};
+				if let Some(value) = token_info.value {
+					// if remote issue mapped token failed, this fund need to transfer token back
+					// to the user. The balance always comes from the user's locked currency while
+					// calling the dispatch call `lock_and_remote_issue`.
+					// This transfer will always successful except some extreme scene, since the
+					// user must lock some currency first, then this transfer can be triggered.
+					let _ = T::RingCurrency::transfer(
+						&Self::pallet_account_id(),
+						&user,
+						value.low_u128().unique_saturated_into(),
+						AllowDeath,
+					);
+				}
+			}
+			Self::deposit_event(Event::TokenLockedConfirmed(token, user, result));
+			return 1;
```

Comment:
We should not return a const weight number, better to record the read and write times in this function. A example like this:

https://github.com/paritytech/parity-bridges-common/blob/master/modules/token-swap/src/lib.rs#L484

Reply:
Here, we use a const low number to ensure the caller not failed.
Furthermore, we need a more accurate calculation of this weight. Because the caller restricts this weight. 
Firstly we should not do so many things in the callback.
Secondly, we need to give an accurate weight of this callback.

Reply:
Yeap, there are some weight limits for a callback function, we can create an issue about this and fix it later.